bundles/org.openhab.binding.solarwatt/src/main/java/org/openhab/binding/solarwatt/internal/domain/model/Location.java

```diff
@@ -0,0 +1,230 @@
+        this.addWattQuantity(CHANNEL_POWER_CONSUMED, deviceDTO);
+        this.addWattQuantity(CHANNEL_POWER_PRODUCED, deviceDTO);
+        this.addWattQuantity(CHANNEL_POWER_OUT, deviceDTO);
+        this.addWattHourQuantity(CHANNEL_WORK_BUFFERED, deviceDTO);
+        this.addWattHourQuantity(CHANNEL_WORK_SELF_CONSUMED, deviceDTO);
+        this.addWattHourQuantity(CHANNEL_WORK_SELF_SUPPLIED, deviceDTO);
+        this.addWattHourQuantity(CHANNEL_WORK_CONSUMED_FROM_GRID, deviceDTO);
+        this.addWattHourQuantity(CHANNEL_WORK_CONSUMED_FROM_STORAGE, deviceDTO);
+        this.addWattHourQuantity(CHANNEL_WORK_CONSUMED, deviceDTO);
+        this.addWattHourQuantity(CHANNEL_WORK_PRODUCED, deviceDTO);
+        this.addWattHourQuantity(CHANNEL_WORK_OUT, deviceDTO);
+
+        // not necessary for a dashboard, so marked as advanced
+        this.addWattQuantity(CHANNEL_POWER_BUFFERED_FROM_GRID, deviceDTO, true);
+        this.addWattQuantity(CHANNEL_POWER_BUFFERED_FROM_PRODUCERS, deviceDTO, true);
+        this.addWattQuantity(CHANNEL_POWER_CONSUMED_FROM_PRODUCERS, deviceDTO, true);
+        this.addWattQuantity(CHANNEL_POWER_IN, deviceDTO, true);
+        this.addWattQuantity(CHANNEL_POWER_OUT_FROM_PRODUCERS, deviceDTO, true);
+        this.addWattQuantity(CHANNEL_POWER_OUT_FROM_STORAGE, deviceDTO, true);
+        this.addWattQuantity(CHANNEL_POWER_RELEASED, deviceDTO, true);
+        this.addWattHourQuantity(CHANNEL_WORK_BUFFERED_FROM_GRID, deviceDTO, true);
+        this.addWattHourQuantity(CHANNEL_WORK_BUFFERED_FROM_PRODUCERS, deviceDTO, true);
+        this.addWattHourQuantity(CHANNEL_WORK_CONSUMED_FROM_PRODUCERS, deviceDTO, true);
+        this.addWattHourQuantity(CHANNEL_WORK_OUT_FROM_PRODUCERS, deviceDTO, true);
+        this.addWattHourQuantity(CHANNEL_WORK_OUT_FROM_STORAGE, deviceDTO, true);
+        this.addWattHourQuantity(CHANNEL_WORK_RELEASED, deviceDTO, true);
+
+        // calculate direct consumption for display purposes
+        this.calculateDifferenceToState(CHANNEL_POWER_SELF_CONSUMED.getChannelName(),
+                CHANNEL_POWER_BUFFERED.getChannelName(), CHANNEL_POWER_DIRECT_CONSUMED.getChannelName(), "energy");
+        this.calculateDifferenceToState(CHANNEL_WORK_SELF_CONSUMED.getChannelName(),
+                CHANNEL_WORK_BUFFERED.getChannelName(), CHANNEL_WORK_DIRECT_CONSUMED.getChannelName(), "energy");
+
+        // read IdDevicesMap to find out which devices are located/metered where
+        // to get the unmetered consumption we need to take Location.(Work|Power)Consumed
+        // and subtract the (Work|Power)(AC)In of the OUTER_CONSUMERs
+        try {
+            JsonObject rawDevicesMap = deviceDTO.getJsonObjectFromTag("IdDevicesMap");
+            Gson gson = new GsonBuilder().create();
+            this.devicesMap = gson.fromJson(rawDevicesMap, IdDevicesMap.class);
+        } catch (Exception ex) {
+            this.devicesMap = null;
+            this.logger.error("Could not read IdDevicesMap", ex);
```

Comment:
and here, please do a search and find all of these.

Reply:
Reduced loglevels.